lib/fluent/supervisor.rb

```diff
@@ -806,6 +807,53 @@ def configure(supervisor: false)
 
       $log.info :supervisor, 'parsing config file is succeeded', path: @config_path
 
+      # merge additional configuration
+      config_type = if File.extname(@config_path) == '.yaml' || File.extname(@config_path) == '.yml'
+                      :yaml
+                    else
+                      nil
+                    end
```

Comment:
As the file extensions are checked some times, I think it is betther to create a method, like:

```ruby
def config_type(config_path)
  if File.extname(config_path) == '.yaml' || File.extname(config_path) == '.yml'
    :yaml
  else
    nil
  end
end


  # merge additional configuration
  config_type = config_type(@config_path)

  ...

    if config_type == :yaml
      if config_type(path) != :yaml
        $log.warn :supervisor, 'skip loading configuration file', path: path
        next
      end
...
```

Reply:
Thanks, I've took a bit differenct approach.